Cargo.toml

```diff
@@ -16,4 +16,10 @@ keywords = ["data-structures"]
 readme = "README.md"
 
 [features]
-nightly = []
+nightly = ["serde"]
+
+[dependencies]
+serde = { optional = true, version = "^0.7" }
+
+[dev-dependencies]
+serde_json = "^0.7"
```

Comment:
Hmm you're right. Do you have a preference for how to solve this? Cargo support for optional dev dependencies is filed at https://github.com/rust-lang/cargo/issues/1596. Until then, we could make `cargo test` run the serde tests, or stick to the current behavior of running them only for `cargo test --features serde` (despite downloading and compiling serde).